### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by dhenneke**
This PR aligns the code formatting to the other projects to make it easier to work with the code.

### CODE REVIEW

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Thank you for your contribution @dhenneke . Changes do basically look good :+1:  Only some smaller notes.

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/build
```
Comment (weeman1337) [inline .prettierignore:1]: This should be based on the project's gitignore. It looks like it is copied from another project.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,43 +1,37 @@
...
+                // Turn off things we're not concerned about here
+                "object-curly-spacing": "off",
+                "array-bracket-spacing": "off",
+                "quote-props": "off", // "be sensible"
+                "@typescript-eslint/no-explicit-any": "off", // handled at review time
 
-            // We want to go a bit long for clarity
-            "max-len": ["error", {"code": 160}],
+                // We want to go a bit long for clarity
+                "max-len": ["error", { code: 160 }],
```
[the remark this answers is not in the document]
Comment (dhenneke) [reply .eslintrc.js]: I removed id.

File: .prettierignore (hunk 1 above)
Comment (dhenneke) [reply .prettierignore]: Yes I copied it from element-web 😅. I now replaced it with the contents of the `.gitignore`.

**Comment by weeman1337** [comment]
I am trying to figure out what to do about Node 14: https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$5Xdul9sib5y7FWTxJXoQhQ3wWt9YMiscF56oOGjnzYk?via=matrix.org&via=element.io&via=envs.net

**Comment by weeman1337** [comment]
> I added `eslint-plugin-unicorn` because `eslint` complained that it wasn't installed.

Yep. It is a peer dependency of [eslint-plugin-matrix-org](https://github.com/matrix-org/eslint-plugin-matrix-org/blob/main/package.json#L33) . Depending on the configuration it has to be installed.

**Comment by dhenneke** [comment]
I rebased the branch (the formatting changes made it hard to update it) and also upgraded the pipeline to use matrix builds for node 18 & latest. If you prefer, I can also move the pipeline part to a separate PR.

**Comment by weeman1337** [comment]
The workflows are still failing. @dhenneke can you use the same as in https://github.com/matrix-org/matrix-react-sdk/blob/develop/.github/workflows/static_analysis.yaml#L70 etc.?

**Comment by dhenneke** [comment]
Yes I will update the actions later 👍

File: package.json [hunk 3]
```diff
@@ -1,62 +1,65 @@
...
+        "LICENSE"
+    ],
+    "scripts": {
+        "clean": "rimraf lib",
+        "build": "yarn clean && yarn build:compile && yarn build:types",
+        "build:types": "tsc -p ./tsconfig.build.json --emitDeclarationOnly",
+        "build:compile": "babel -d lib --verbose --extensions \".ts,.tsx\" src",
+        "start": "tsc -p ./tsconfig.build.json -w",
+        "test": "jest",
+        "lint": "eslint src test && tsc --noEmit && prettier --check",
```
Comment (weeman1337) [inline package.json:24]: Prettier needs a argument what to check:

```suggestion
        "lint": "eslint src test && tsc --noEmit && prettier --check .",
```

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Getting closer :slightly_smiling_face: Can you fix the lint error and the script? If the new workflows are green we can switch the required workflows and merge it.

**Review by weeman1337 — APPROVED** [review]
Looks good now, thanks :+1:

**Comment by richvdh** [comment]
@weeman1337 will you merge this?

**Comment by weeman1337** [comment]
> @weeman1337 will you merge this?

I've just learned, that „You’re not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge this pull request.“ @richvdh can you merge it?

**Pull request merged**